### PR TITLE
Set 6 Codacy issues

### PR DESCRIPTION
**Pull request by aayusharora**

### CODE REVIEW

File: src/appComponents/sponsors/SponsorsController.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ sponsorsModule.controller('SponsorsController',
         var sc = this;
```
Comment (juslee) [inline src/appComponents/sponsors/SponsorsController.js]: Codacy Issue found: [Unexpected `this`.](https://www.codacy.com/app/triplez/open-event-webapp/file/2417978710/issues/source?bid=3233857&fileBranchId=3237740#l19)

File: src/appComponents/tracks/TracksController.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ tracksModule.controller('TracksController',
         function($mdDialog, $sessionStorage, $rootScope, ApiJsonFactory) {
 		var tc = this;
         if ( $sessionStorage.tracks === null ||
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [There should be no spaces inside this paren.](https://www.codacy.com/app/triplez/open-event-webapp/file/2417978711/issues/source?bid=3233857&fileBranchId=3237740#l19)

File: src/appComponents/tracks/TracksController.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ tracksModule.controller('TracksController',
...
-            typeof($sessionStorage.tracks) == 'undefined')
+            typeof($sessionStorage.tracks) === 'undefined')
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Unary word operator "typeof" must be followed by whitespace.](https://www.codacy.com/app/triplez/open-event-webapp/file/2417978711/issues/source?bid=3233857&fileBranchId=3237740#l20)

File: src/appComponents/sponsors/SponsorsController.js (hunk 1 above)
Comment (juslee) [inline src/appComponents/sponsors/SponsorsController.js]: Codacy Issue found: [Unexpected `this`.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418082362/issues/source?bid=3233857&fileBranchId=3237740#l19)

File: src/appComponents/tracks/TracksController.js [hunk 4]
```diff
@@ -26,15 +28,15 @@ tracksModule.controller('TracksController',
         
 
         if (tc.Tracks.length === 0) {
-            ApiJsonFactory.getJson('tracks')
+            ApiJsonFactory.getJson("tracks")
                 .then(function (response) {
                     tc.Tracks = response.data.tracks;
                     $sessionStorage.tracks = tc.Tracks;
                 }, function (error) {
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 18 space characters but found 20.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418082363/issues/source?bid=3233857&fileBranchId=3237740#l36)

File: src/appComponents/tracks/TracksController.js [hunk 5]
```diff
@@ -2,55 +2,57 @@
...
-        if (tc.Tracks.length === 0) {
-            ApiJsonFactory.getJson('tracks')
+        if (tsc.Tracks.length === 0) {
+            ApiJsonFactory.getJson("tracks")
                 .then(function (response) {
-                    tc.Tracks = response.data.tracks;
-                    $sessionStorage.tracks = tc.Tracks;
+                    tsc.Tracks = response.data.tracks;
+                    $sessionStorage.tracks = tsc.Tracks;
                 }, function (error) {
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: ['error' is defined but never used.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418086922/issues/source?bid=3233857&fileBranchId=3237740#l35)

File: src/appComponents/tracks/TracksController.js [hunk 6]
```diff
@@ -2,55 +2,57 @@
...
-	['$mdDialog','$sessionStorage', '$rootScope', 'ApiJsonFactory',
-        function($mdDialog, $sessionStorage, $rootScope, ApiJsonFactory) {
-		var tc = this;
-        if ( $sessionStorage.tracks === null ||
-            typeof($sessionStorage.tracks) == 'undefined')
+tracksModule.controller("TracksController", ["$mdDialog","$sessionStorage", "$rootScope", "ApiJsonFactory",
+        function ($mdDialog, $sessionStorage, $rootScope, ApiJsonFactory) {
+          var tsc = this;
+
+          if ($sessionStorage.tracks === null ||
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Trailing spaces not allowed.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418086922/issues/source?bid=3233857&fileBranchId=3237740#l21)

File: src/appComponents/tracks/TracksController.js [hunk 7]
```diff
@@ -2,55 +2,57 @@
...
         }
-        tc.Tracks = $sessionStorage.tracks;
-        tc.Sessions = $sessionStorage.sessions;
+        tsc.Tracks = $sessionStorage.tracks;
+        tsc.Sessions = $sessionStorage.sessions;
         
 
-        if (tc.Tracks.length === 0) {
-            ApiJsonFactory.getJson('tracks')
+        if (tsc.Tracks.length === 0) {
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 10 space characters but found 8.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418086922/issues/source?bid=3233857&fileBranchId=3237740#l30)

File: src/appComponents/tracks/TracksController.js [hunk 8]
```diff
@@ -2,55 +2,57 @@
...
-            typeof($sessionStorage.tracks) == 'undefined')
+tracksModule.controller("TracksController", ["$mdDialog","$sessionStorage", "$rootScope", "ApiJsonFactory",
+        function ($mdDialog, $sessionStorage, $rootScope, ApiJsonFactory) {
+          var tsc = this;
+
+          if ($sessionStorage.tracks === null || 
+            typeof $sessionStorage.tracks === "undefined")
         {
             $sessionStorage.tracks = [];
         }
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 10 space characters but found 8.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418086922/issues/source?bid=3233857&fileBranchId=3237740#l25)

File: src/appComponents/tracks/TracksController.js [hunk 9]
```diff
@@ -2,55 +2,57 @@
...
-                    $sessionStorage.tracks = tc.Tracks;
+                    tsc.Tracks = response.data.tracks;
+                    $sessionStorage.tracks = tsc.Tracks;
                 }, function (error) {
-                    console.error(error);
+                    //console.error(error);
                 });
 
-            ApiJsonFactory.getJson('sessions')
+            ApiJsonFactory.getJson("sessions")
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 10 space characters but found 12.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418086922/issues/source?bid=3233857&fileBranchId=3237740#l39)

File: src/appComponents/tracks/TracksController.js (hunk 5 above)
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Using "FunctionExpression" is not allowed.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418086922/issues/source?bid=3233857&fileBranchId=3237740#l35)

Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Block must be padded by blank lines.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418086922/issues/source?bid=3233857&fileBranchId=3237740#l35)

**Comment by juslee** [comment]
Fix the issues identified above @aayusharora

File: src/appComponents/tracks/TracksController.js (hunk 5 above)
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: ['error' is defined but never used.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418113389/issues/source?bid=3233857&fileBranchId=3237740#l35)

File: src/appComponents/tracks/TracksController.js (hunk 6 above)
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Trailing spaces not allowed.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418113389/issues/source?bid=3233857&fileBranchId=3237740#l21)

File: src/appComponents/tracks/TracksController.js (hunk 5 above)
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: ['error' is defined but never used.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418113389/issues/source?bid=3233857&fileBranchId=3237740#l35)

File: src/appComponents/tracks/TracksController.js (hunk 7 above)
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 10 space characters but found 8.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418113389/issues/source?bid=3233857&fileBranchId=3237740#l30)

File: src/appComponents/tracks/TracksController.js (hunk 6 above)
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Trailing spaces not allowed.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418113389/issues/source?bid=3233857&fileBranchId=3237740#l21)

File: src/appComponents/tracks/TracksController.js (hunk 8 above)
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 10 space characters but found 8.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418113389/issues/source?bid=3233857&fileBranchId=3237740#l25)

File: src/appComponents/tracks/TracksController.js (hunk 7 above)
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 10 space characters but found 8.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418113389/issues/source?bid=3233857&fileBranchId=3237740#l30)

File: src/appComponents/tracks/TracksController.js (hunk 9 above)
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 10 space characters but found 12.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418113389/issues/source?bid=3233857&fileBranchId=3237740#l39)

File: src/appComponents/tracks/TracksController.js (hunk 8 above)
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 10 space characters but found 8.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418113389/issues/source?bid=3233857&fileBranchId=3237740#l25)

File: src/appComponents/tracks/TracksController.js (hunk 5 above)
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Using "FunctionExpression" is not allowed.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418113389/issues/source?bid=3233857&fileBranchId=3237740#l35)

File: src/appComponents/tracks/TracksController.js (hunk 9 above)
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 10 space characters but found 12.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418113389/issues/source?bid=3233857&fileBranchId=3237740#l39)

File: src/appComponents/tracks/TracksController.js (hunk 5 above)
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Block must be padded by blank lines.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418113389/issues/source?bid=3233857&fileBranchId=3237740#l35)

Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Using "FunctionExpression" is not allowed.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418113389/issues/source?bid=3233857&fileBranchId=3237740#l35)

Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Block must be padded by blank lines.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418113389/issues/source?bid=3233857&fileBranchId=3237740#l35)

**Comment by aayusharora** [comment]
@juslee , Not able to add a line note here.

File: src/appComponents/tracks/TracksController.js (hunk 9 above)
Comment (aayusharora) [inline src/appComponents/tracks/TracksController.js]: Its already 10

File: src/appComponents/tracks/TracksController.js [hunk 10]
```diff
@@ -1,56 +1,65 @@
...
-                }, function (error) {
-                    console.error(error);
-                });
+        if (tsc.Tracks.length === 0) {
+
+          ApiJsonFactory.getJson("tracks")
+            .then(function (response) {
+                tsc.Tracks = response.data.tracks;
+                $sessionStorage.tracks = tsc.Tracks;
+            }, function error (error) {
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: ['error' is defined but never used.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418144097/issues/source?bid=3233857&fileBranchId=3237740#l39)

File: src/appComponents/tracks/TracksController.js [hunk 11]
```diff
@@ -1,56 +1,65 @@
...
-                    $sessionStorage.sessions = tc.Sessions;
-                }, function (error) {
-                    console.error(error);
+          ApiJsonFactory.getJson("sessions")
+            .then(function (response) {
+
+                tsc.Sessions = response.data.sessions;
+                $sessionStorage.sessions = tsc.Sessions;
+
+            }, function (error) {
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: ['error' is defined but never used.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418144097/issues/source?bid=3233857&fileBranchId=3237740#l49)

File: src/appComponents/tracks/TracksController.js [hunk 12]
```diff
@@ -1,56 +1,65 @@
...
+                    //console.error(error);
                 });
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 12 space characters but found 16.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418144097/issues/source?bid=3233857&fileBranchId=3237740#l51)

File: src/appComponents/tracks/TracksController.js [hunk 13]
```diff
@@ -1,56 +1,65 @@
 /**
  * Created by championswimmer on 29/5/15.
  */
+/* exported error */
 
-var tracksModule = angular.module('oe.tracks', ['ui.router']);
+const tracksModule = angular.module("oe.tracks", ["ui.router"]);
+
+tracksModule.config(["$stateProvider", function($stateProvider) {
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Identifier '$stateProvider' does not match the pattern '^[a-z]+([A-Z][a-z]+)*$'.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418144097/issues/source?bid=3233857&fileBranchId=3237740#l8)

File: src/appComponents/tracks/TracksController.js [hunk 14]
```diff
@@ -1,56 +1,65 @@
...
-                .then(function (response) {
-                    tc.Tracks = response.data.tracks;
-                    $sessionStorage.tracks = tc.Tracks;
-                }, function (error) {
-                    console.error(error);
-                });
+        if (tsc.Tracks.length === 0) {
+
+          ApiJsonFactory.getJson("tracks")
+            .then(function (response) {
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Unexpected function expression.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418144097/issues/source?bid=3233857&fileBranchId=3237740#l36)

File: src/appComponents/tracks/TracksController.js [hunk 15]
```diff
@@ -1,56 +1,65 @@
...
-tracksModule.controller('TracksController', 
-	['$mdDialog','$sessionStorage', '$rootScope', 'ApiJsonFactory',
-        function($mdDialog, $sessionStorage, $rootScope, ApiJsonFactory) {
-		var tc = this;
-        if ( $sessionStorage.tracks === null ||
-            typeof($sessionStorage.tracks) == 'undefined')
-        {
+tracksModule.controller("TracksController", ["$mdDialog","$sessionStorage", "$rootScope", "ApiJsonFactory",
+        function ($mdDialog, $sessionStorage, $rootScope, ApiJsonFactory) {
+          var tsc = this;
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Unexpected `this`.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418144097/issues/source?bid=3233857&fileBranchId=3237740#l20)

File: src/appComponents/tracks/TracksController.js [hunk 16]
```diff
@@ -1,56 +1,65 @@
...
-            ApiJsonFactory.getJson('sessions')
-                .then(function (response) {
-                    tc.Sessions = response.data.sessions;
-                    $sessionStorage.sessions = tc.Sessions;
-                }, function (error) {
-                    console.error(error);
+          ApiJsonFactory.getJson("sessions")
+            .then(function (response) {
+
+                tsc.Sessions = response.data.sessions;
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 18 space characters but found 20.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418144097/issues/source?bid=3233857&fileBranchId=3237740#l46)

File: src/appComponents/tracks/TracksController.js [hunk 17]
```diff
@@ -1,56 +1,65 @@
...
+                    //console.error(error);
                 });
+
         }
 
-        tc.showSession = function(track, event) {
+        tsc.showSession = function(track, event) {
             //singleTrack = track;
             $mdDialog.track = {
                 singleTrack: track
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Unquoted property `singleTrack` found.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418144097/issues/source?bid=3233857&fileBranchId=3237740#l58)

File: src/appComponents/tracks/TracksController.js [hunk 18]
```diff
@@ -1,56 +1,65 @@
...
+         {
+
             $sessionStorage.tracks = [];
-        }
-        tc.Tracks = $sessionStorage.tracks;
-        tc.Sessions = $sessionStorage.sessions;
+
+          }
+        tsc.Tracks = $sessionStorage.tracks;
+        tsc.Sessions = $sessionStorage.sessions;
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Trailing spaces not allowed.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418144097/issues/source?bid=3233857&fileBranchId=3237740#l31)

File: src/appComponents/tracks/TracksController.js [hunk 19]
```diff
@@ -1,56 +1,65 @@
...
+
+            }, function (error) {
+                    //console.error(error);
                 });
+
         }
 
-        tc.showSession = function(track, event) {
+        tsc.showSession = function(track, event) {
             //singleTrack = track;
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected space or tab after "//" in comment.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418144097/issues/source?bid=3233857&fileBranchId=3237740#l56)

File: src/appComponents/tracks/TracksController.js [hunk 20]
```diff
@@ -1,100 +1,120 @@
...
+                    //console.error(error);
+            });
+
+          ApiJsonFactory.getJson("sessions")
+            .then(function (response) {
+
+            tsc.Sessions = response.data.sessions;
+            $sessionStorage.sessions = tsc.Sessions;
+
+            }, function (error) {
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Unexpected function expression.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418394575/issues/source?bid=3233857&fileBranchId=3237740#l50)

File: src/appComponents/tracks/TracksController.js [hunk 21]
```diff
@@ -1,100 +1,120 @@
 /**
  * Created by championswimmer on 29/5/15.
  */
+/* exported error */
 
-var tracksModule = angular.module('oe.tracks', ['ui.router']);
+const tracksModule = angular.module("oe.tracks", ["ui.router"]);
+
+tracksModule.config(["$stateProvider", function($stateProvider) {
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Identifier '$stateProvider' does not match the pattern '^[a-z]+([A-Z][a-z]+)*$'.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418394575/issues/source?bid=3233857&fileBranchId=3237740#l8)

File: src/appComponents/tracks/TracksController.js [hunk 22]
```diff
@@ -1,100 +1,120 @@
...
-            $mdDialog.show({
-                controller: 'TrackDialogController',
-                templateUrl: 'appComponents/tracks/trackdialog.html',
+        tsc.showSession = function(track, event) {
+            // singleTrack = track;
+          $mdDialog.track = {
+                "singleTrack": track
+          };
+          $mdDialog.show({
+                controller: "TrackDialogController",
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Unquoted property `controller` found.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418394575/issues/source?bid=3233857&fileBranchId=3237740#l62)

File: src/appComponents/tracks/TracksController.js [hunk 23]
```diff
@@ -1,100 +1,120 @@
...
-tracksModule.controller('TracksController', 
-	['$mdDialog','$sessionStorage', '$rootScope', 'ApiJsonFactory',
-        function($mdDialog, $sessionStorage, $rootScope, ApiJsonFactory) {
-		var tc = this;
-        if ( $sessionStorage.tracks === null ||
-            typeof($sessionStorage.tracks) == 'undefined')
-        {
+tracksModule.controller("TracksController", ["$mdDialog","$sessionStorage", "$rootScope", "ApiJsonFactory",
+        function ($mdDialog, $sessionStorage, $rootScope, ApiJsonFactory) {
+          var tsc = this;
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Unexpected `this`.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418394575/issues/source?bid=3233857&fileBranchId=3237740#l20)

File: src/appComponents/tracks/TracksController.js (hunk 20 above)
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418394575/issues/source?bid=3233857&fileBranchId=3237740#l50)

File: src/appComponents/tracks/TracksController.js [hunk 24]
```diff
@@ -1,100 +1,120 @@
...
+         {
+
             $sessionStorage.tracks = [];
-        }
-        tc.Tracks = $sessionStorage.tracks;
-        tc.Sessions = $sessionStorage.sessions;
+
+          }
+        tsc.Tracks = $sessionStorage.tracks;
+        tsc.Sessions = $sessionStorage.sessions;
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Trailing spaces not allowed.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418394575/issues/source?bid=3233857&fileBranchId=3237740#l31)

File: src/appComponents/tracks/TracksController.js [hunk 25]
```diff
@@ -1,100 +1,120 @@
...
-                controller: 'TrackDialogController',
-                templateUrl: 'appComponents/tracks/trackdialog.html',
+        tsc.showSession = function(track, event) {
+            // singleTrack = track;
+          $mdDialog.track = {
+                "singleTrack": track
+          };
+          $mdDialog.show({
+                controller: "TrackDialogController",
+                templateUrl: "appComponents/tracks/trackdialog.html",
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Unquoted property `templateUrl` found.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418394575/issues/source?bid=3233857&fileBranchId=3237740#l63)

File: src/appComponents/tracks/TracksController.js [hunk 26]
```diff
@@ -1,100 +1,119 @@
...
-                }, function (error) {
-                    console.error(error);
-                });
+          ApiJsonFactory.getJson("tracks")
+            .then(function (response) {
+
+            tsc.Tracks = response.data.tracks;
+            $sessionStorage.tracks = tsc.Tracks;
+                
+            }, function error (error) {
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: ['error' is defined but never used.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418442828/issues/source?bid=3233857&fileBranchId=3237740#l39)

File: src/appComponents/tracks/TracksController.js [hunk 27]
```diff
@@ -1,100 +1,119 @@
...
-                    $sessionStorage.sessions = tc.Sessions;
-                }, function (error) {
-                    console.error(error);
+          ApiJsonFactory.getJson("sessions")
+            .then(function (response) {
+
+            tsc.Sessions = response.data.sessions;
+            $sessionStorage.sessions = tsc.Sessions;
+
+            }, function (error) {
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: ['error' is defined but never used.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418442828/issues/source?bid=3233857&fileBranchId=3237740#l49)

File: src/appComponents/tracks/TracksController.js [hunk 28]
```diff
@@ -1,100 +1,119 @@
...
+                    //console.error(error);
                 });
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 12 space characters but found 16.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418442828/issues/source?bid=3233857&fileBranchId=3237740#l51)

File: src/appComponents/tracks/TracksController.js [hunk 29]
```diff
@@ -1,100 +1,119 @@
 /**
  * Created by championswimmer on 29/5/15.
  */
+/* exported error */
 
-var tracksModule = angular.module('oe.tracks', ['ui.router']);
+const tracksModule = angular.module("oe.tracks", ["ui.router"]);
+
+tracksModule.config(["$stateProvider", function($stateProvider) {
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Identifier '$stateProvider' does not match the pattern '^[a-z]+([A-Z][a-z]+)*$'.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418442828/issues/source?bid=3233857&fileBranchId=3237740#l8)

File: src/appComponents/tracks/TracksController.js [hunk 30]
```diff
@@ -1,100 +1,119 @@
...
-                parent: angular.element(document.body),
-                targetEvent: event,
+        tsc.showSession = function(track, event) {
+            // singleTrack = track;
+          $mdDialog.track = {
+                "singleTrack": track
+          };
+          $mdDialog.show({
+                "controller": "TrackDialogController",
+                "templateUrl": "appComponents/tracks/trackdialog.html",
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 12 space characters but found 16.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418442828/issues/source?bid=3233857&fileBranchId=3237740#l62)

File: src/appComponents/tracks/TracksController.js [hunk 31]
```diff
@@ -1,100 +1,119 @@
...
-            };
-            $mdDialog.show({
-                controller: 'TrackDialogController',
-                templateUrl: 'appComponents/tracks/trackdialog.html',
-                parent: angular.element(document.body),
-                targetEvent: event,
+        tsc.showSession = function(track, event) {
+            // singleTrack = track;
+          $mdDialog.track = {
+                "singleTrack": track
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Expected indentation of 12 space characters but found 16.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418442828/issues/source?bid=3233857&fileBranchId=3237740#l58)

File: src/appComponents/tracks/TracksController.js [hunk 32]
```diff
@@ -1,100 +1,119 @@
...
-tracksModule.controller('TracksController', 
-	['$mdDialog','$sessionStorage', '$rootScope', 'ApiJsonFactory',
-        function($mdDialog, $sessionStorage, $rootScope, ApiJsonFactory) {
-		var tc = this;
-        if ( $sessionStorage.tracks === null ||
-            typeof($sessionStorage.tracks) == 'undefined')
-        {
+tracksModule.controller("TracksController", ["$mdDialog","$sessionStorage", "$rootScope", "ApiJsonFactory",
+        function ($mdDialog, $sessionStorage, $rootScope, ApiJsonFactory) {
+          var tsc = this;
```
Comment (juslee) [inline src/appComponents/tracks/TracksController.js]: Codacy Issue found: [Unexpected `this`.](https://www.codacy.com/app/triplez/open-event-webapp/file/2418442828/issues/source?bid=3233857&fileBranchId=3237740#l20)

**Pull request merged**